components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/internal/DefaultServiceURLBuilder.java

```diff
@@ -0,0 +1,447 @@
+         */
+        @Override
+        public String getPath() {
+
+            return urlPath;
+        }
+
+        /**
+         * Returns the parameter value when the key is provided.
+         *
+         * @param key Key of the parameter.
+         * @return The value of the parameter.
+         */
+        @Override
+        public String getParameter(String key) {
+
+            return parameters.get(key);
+        }
+
+        /**
+         * Returns a map of the parameters.
+         *
+         * @return The parameters.
+         */
+        @Override
+        public Map<String, String> getParameters() {
+
+            return Collections.unmodifiableMap(parameters);
+        }
+
+        /**
+         * Returns decoded fragment from the url.
+         *
+         * @return The decoded fragment.
+         */
+        @Override
+        public String getFragment() {
+
+            return fragment;
+        }
+
+        /**
+         * Returns the tenant domain of the service URL.
+         *
+         * @return The tenant domain.
+         */
+        @Override
+        public String getTenantDomain() {
+
+            return tenantDomain;
+        }
+
+        /**
+         * Concatenate the protocol, host name, port, proxy context path, web context root, url context, query params
+         * and the fragment to return the absolute URL.
+         *
+         * @return The absolute URL from the Service URL instance.
+         */
+        @Override
+        public String getAbsoluteURL() {
+
+            return absoluteUrl;
+        }
+
+        /**
+         * Concatenate the url context, query params and the fragment to return the relative URL.
+         *
+         * @return The relative URL from the Service URL instance.
+         */
+        @Override
+        public String getRelativeURL() {
+
+            return relativeUrl;
+        }
+
+        private String fetchAbsoluteUrl() throws ServiceURLException {
+
+            StringBuilder absoluteUrl = new StringBuilder();
+            absoluteUrl.append(protocol).append("://");
+
+            absoluteUrl.append(hostName.toLowerCase());
+
+            // If it's well known HTTPS port, skip adding port.
+            if (port != IdentityCoreConstants.DEFAULT_HTTPS_PORT) {
+                absoluteUrl.append(":").append(port);
+            }
+
+            if (StringUtils.isNotBlank(urlPath)) {
+                if (urlPath.trim().charAt(0) != '/' && urlPath.trim().charAt(0) != '?') {
+                    absoluteUrl.append("/").append(urlPath.trim());
+                } else {
+                    absoluteUrl.append(urlPath.trim());
+                }
+            }
+            String resolvedParamsString = getResolvedParamString(parameters);
+            try {
+                appendParamsToUri(absoluteUrl, resolvedParamsString, "?");
+                appendParamsToUri(absoluteUrl, fragment, "#");
+
+            } catch (URLBuilderException e) {
```

Comment:
Why we need to wrap this and throw ?
Still this relates to the same thing right. Let's throw directly one type